### PR TITLE
Bug fixes in LocalNetwork and batch_eth_mnist

### PR DESCRIPTION
**Pull request by Rushil-Chakra**
Previously, the model would crash after each update_interval with plotting on because the labels tensor would be cleared but step would not reset to 0

### CODE REVIEW

**Review by coopersigrist — COMMENTED** [review]
This looks good to me but the check failed, this might have issues on edge cases

**Comment by Rushil-Chakra** [comment]
It looks like the check fail on network.py when calling save(). I'm assuming plotting doesn't affect the network so I think it's an unrelated problem? Also the previous pull request seems to be having the exact same issue.

**Comment by Rushil-Chakra** [comment]
Pytorch documentation recommends using torch.save(self.state_dict()) rather than the torch.save(self) for network.save(), as the latter seems more sensitive and prone to breaking.
https://pytorch.org/docs/master/notes/serialization.html

**Comment by coopersigrist** [comment]
@djsaunde what do you think? would it be fine to make the change rushil is suggesting?

**Comment by Rushil-Chakra** [comment]
I added a change to LocalNetwork, the input layer wasn't taking in the input_shape network parameter. The input is also assumed to have 3 dimensions, channels, width, and height.
